src/server/network/drain.c

```
@@ -0,0 +1,279 @@
+{
+    if (out_size < BC_MAX_PACKET_SIZE) return 0;
+
+    int pos = BC_DRAIN_HEADER_LEN;  /* reserve 2-byte header */
+    int msg_count = 0;
+    int promoted = 0;
+
+    /* --- Pass 1: priority-reliable, retx < 3 -- pack until won't-fit/cap --- */
+    {
+        int scanned = 0;
+        int total = d->priority.count;
+        while (scanned < total) {
```

Clang (14.0.6)

[note] +1, including nesting penalty of 0, nesting level increased to 1

(clang)